### PR TITLE
[INT-194] Only open slack socket on the persistent server

### PR DESCRIPTION
**Pull request by archessmn**
This should hopefully reduce the logged messages from the slack connection restarting.

### CODE REVIEW

File: lib/slack/slackApiConnection.ts [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-}
-
-let app = global.slack;
-
 export const isSlackEnabled = env.SLACK_ENABLED === "true";
 
-if (!app && isSlackEnabled) {
-  app = global.slack = new App({
+async function slackApiConnection() {
+  return new App({
```
Comment (markspolakovs) [inline lib/slack/slackApiConnection.ts:7]: Are we sure there aren't places where this would get called twice per request?

(Also, why is this change necessary? Surely the persistent server has its own memory space so can have its own global slack client cache. If the only reason is to avoid calling `app.start()`, you could pass in a `shouldStartApp = false` parameter.)

**Review by markspolakovs — CHANGES_REQUESTED** [review]
Makes sense, but I'm not sure reverting the global cache is correct.

File: lib/slack/slackApiConnection.ts (hunk 1 above)
Comment (archessmn) [reply lib/slack/slackApiConnection.ts:7]: I didn't think that the global cache actually did much but am happy to revert it if that is the case. Having `app.start()` separate is preferable, as the boltjs example code shows listeners etc being declared before starting the app, which wouldn't be possible if the app was started before being returned.

Comment (markspolakovs) [reply lib/slack/slackApiConnection.ts:7]: The main benefit of the global cache is not creating multiple `App`s if `slackApiConnection()` is called more than once per request - I'm not sure if we do it (we shouldn't), but it's a decent precaution. It also saves on some overhead between requests (I'm not sure how expensive an `App` is to create).

**Review by markspolakovs — APPROVED** [review]
